test-app/tests/integration/components/file-input-field-test.gts

```diff
@@ -0,0 +1,274 @@
+
+import {
+  find,
+  render,
+  settled,
+  setupOnerror,
+  triggerEvent,
+} from '@ember/test-helpers';
+import { module, test } from 'qunit';
+
+import FileInputField from '@crowdstrike/ember-toucan-core/components/form/file-input-field';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+
+// https://medium.com/@chrisdmasters/acceptance-testing-file-uploads-in-ember-2-5-1c9c8dbe5368
+function createFile(
+  content = ['Some sample content'],
+  options: Options = { name: '', type: '' }
+) {
+  const { name, type } = options;
+
+  const file = new File(content, name, {
+    type: type ? type : 'text/plain',
+  });
+
+  return file;
+}
+
+type Options = {
+  name: string;
+  type?: string;
+};
+
+module('Integration | Component | FileInputField', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" data-file-input-field />
+    </template>);
+
+    assert.dom('[data-label]').hasText('Label');
+
+    assert
+      .dom('[data-hint]')
+      .doesNotExist(
+        'Expected hint block not to be displayed as a hint was not provided'
+      );
+
+    assert.dom('[data-file-input-field]').hasTagName('input');
+    assert.dom('[data-file-input-field]').hasAttribute('id');
+    assert
+      .dom('[data-file-input-field]')
+      .hasClass('text-titles-and-attributes');
+
+    assert
+      .dom('[data-error]')
+      .doesNotExist(
+        'Expected hint block not to be displayed as an error was not provided'
+      );
+  });
+
+  test('it renders with a hint', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" @hint="Hint text" data-file-input-field />
+    </template>);
+
+    // For the file input field component, the only aria-describedby
+    // value should be the errorId.  This is due to the way the component
+    // is structured, where the label+hint are rendered inside of the
+    // wrapping <label> element
+    const hint = find('[data-hint]');
+
+    assert.dom(hint).hasText('Hint text');
+  });
+
+  test('it renders with an error', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @error="Error text"
+        data-file-input-field
+      />
+    </template>);
+
+    let error = find('[data-error]');
+
+    assert.dom(error).hasText('Error text');
+    assert.dom(error).hasAttribute('id');
+
+    let errorId = error?.getAttribute('id') || '';
+    assert.ok(errorId, 'Expected errorId to be truthy');
+
+    // For the file input field component, the only aria-describedby
+    // value should be the errorId.  This is due to the way the component
+    // is structured, where the label+hint are rendered inside of the
+    // wrapping <label> element
+    let describedby =
+      find('[data-file-input-field]')?.getAttribute('aria-describedby') || '';
```

Comment:
nitpick, but in general I would always use `??` instead of `||` for applying a default value. In this case it does not matter, but for things that e.g. return a boolean it is tricky, as a legit value could be falsey then and overridden by the default value. Like `return this.isValid || true`, this would _always_ be true, even when `this.isValid` returns `false`. But `??` would do the expected thing here. And so for consistency, I would just think of `||` as a boolean or operation, and `??` as "apply a default value". 

Just my way of thinking about this, but feel free to ignore! 😏